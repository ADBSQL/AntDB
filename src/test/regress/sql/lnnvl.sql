set grammar to oracle;
create table lnnvltbl(name varchar(10), id int);
insert into lnnvltbl values(null, 1); 
insert into lnnvltbl values('', 2);
insert into lnnvltbl values ('null',3);
insert into lnnvltbl values('s', 4);
insert into lnnvltbl values ('ss',5);
select * from lnnvltbl where lnnvl(name is not null) order by id;
select * from lnnvltbl where lnnvl(name is null) order by id;
select * from lnnvltbl where lnnvl(name like 's%') order by id;
select * from lnnvltbl where lnnvl(name ='s') order by id;
select * from lnnvltbl where lnnvl(name like 's%') and id <4 order by id;
select * from lnnvltbl where lnnvl(id <2 or id >4) order by id;
select * from lnnvltbl where lnnvl(id in(1,3)) order by id;
select * from lnnvltbl where lnnvl(id between 1 and 3) order by id;
drop table lnnvltbl;
