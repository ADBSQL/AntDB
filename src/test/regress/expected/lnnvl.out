set grammar to oracle;
create table lnnvltbl(name varchar(10), id int);
insert into lnnvltbl values(null, 1); 
insert into lnnvltbl values('', 2);
insert into lnnvltbl values ('null',3);
insert into lnnvltbl values('s', 4);
insert into lnnvltbl values ('ss',5);
select * from lnnvltbl where lnnvl(name is not null) order by id;
 NAME | ID 
------+----
      |  1
      |  2
(2 rows)

select * from lnnvltbl where lnnvl(name is null) order by id;
 NAME | ID 
------+----
 null |  3
 s    |  4
 ss   |  5
(3 rows)

select * from lnnvltbl where lnnvl(name like 's%') order by id;
 NAME | ID 
------+----
      |  1
      |  2
 null |  3
(3 rows)

select * from lnnvltbl where lnnvl(name ='s') order by id;
 NAME | ID 
------+----
      |  1
      |  2
 null |  3
 ss   |  5
(4 rows)

select * from lnnvltbl where lnnvl(name like 's%') and id <4 order by id;
 NAME | ID 
------+----
      |  1
      |  2
 null |  3
(3 rows)

select * from lnnvltbl where lnnvl(id <2 or id >4) order by id;
 NAME | ID 
------+----
      |  2
 null |  3
 s    |  4
(3 rows)

select * from lnnvltbl where lnnvl(id in(1,3)) order by id;
 NAME | ID 
------+----
      |  2
 s    |  4
 ss   |  5
(3 rows)

select * from lnnvltbl where lnnvl(id between 1 and 3) order by id;
 NAME | ID 
------+----
 s    |  4
 ss   |  5
(2 rows)

drop table lnnvltbl;
